src/librustdoc/html/static/main.js

```diff
@@ -54,7 +54,8 @@
     }
 
     function browserSupportsHistoryApi() {
-        return window.history && typeof window.history.pushState === "function";
+        return document.origin != "null" &&
```

Comment:
document.origin doesn't seem to be well-supported: https://developer.mozilla.org/en-US/docs/Web/API/Document/origin